pkg/types/installconfig.go

```diff
@@ -90,6 +99,11 @@ type InstallConfig struct {
 	// ImageContentSources lists sources/repositories for the release-image content.
 	// +optional
 	ImageContentSources []ImageContentSource `json:"imageContentSources,omitempty"`
+
+	// Publish controls how the user facing endpoints of the cluster like the k8s API, OpenShift routes etc. are exposed.
+	// When no strategy is specified, the stategy is `External`.
+	// +optional
+	Publish PublishingStrategy `json:"publish,omitempty"`
```

Comment:
is case sensitivity handled in install config? e.g. are both `publish: external` and `publish: External` valid?

Reply:
Currently I do not have any case checking. I do have verification that it's set to "Internal" or "External" though. I agree we should add case checking and force a value.